### ISSUE TITLE
Docs!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! (and other ways of helping potential contributors)

### ISSUE THREAD

**Issue by one-some** [one-some/KoboldAI-united#28]
second try's the charm

organize things that people seem to struggle with the most/need to use the most higher:

### auto variable syncing rundown
- we can use this to explain how the socket comms work
- overrides on the client side
- aspects abt variables like no sync and stuff

### from-browser generation tour
- (socket -> handler -> generate -> core generate -> raw generate -> hf generate)

### "case study" type things with various diverse features to give people an idea of how different types of things work.
- i like this approach a lot since this is basically how i learned the codebase (copy/paste something similar to what I want to do, edit it from there)
- good at developing internal associations like "oh so this is like X but with Y"-- hopefully makes the docs more "potent" in the sense that even if we don't have too many, people can learn to navigate the codebase. "teach a man to fish" and all that
- topic ideas:
  - token streaming (post token hooks, boolean settings)
  - temperature (warpers, number settings)
  - memory (basic variable syncing)
  - privacy mode/lock (ui/modal/backend handling)
  - "Hide timing information" tweak (tweaks)
  - openai backend (basic backend stuff)
  - hftorch backend (more advanced backend stuff: sampling, streaming, stopping, etc)
  - various themes

### contributing.md
- see https://github.com/henk717/KoboldAI/issues/318
- basically, lets not scare people away. things we need to document here:
  - united vs stable for PRs
  - join the discord to discuss and ask questions (`#new-features-coding` is basically the gold standard for docs as of now)
  - expectations on when to hear back
  - link to rest of docs
  - maybe a q&a of some sort (wtf is a ui2)
- probably dont need to add who to ping for review since pr volume is really slow and we'll see everything with the webhook
- we should also make a roadmap of some sort that isnt word of mouth/vague "2.0 soon"
- we dont really have a global style

### adding docs on each backend socket endpoint
- i imagine it could be annoying to have to check koboldai.js just to get a quick rundown of what something does
